tests/cupy_tests/creation_tests/test_from_data.py

```diff
@@ -466,22 +478,28 @@ def test_asfortranarray_cuda_array_zero_dim_dtype(
 
 class DummyObjectWithCudaArrayInterface(object):
 
-    def __init__(self, a, has_strides=False):
+    def __init__(self, a):
         self.a = a
-        self.has_strides = has_strides
+        self._desc = None
 
     @property
     def __cuda_array_interface__(self):
-        desc = {
-            'shape': self.a.shape,
-            'typestr': self.a.dtype.str,
-            'descr': self.a.dtype.descr,
-            'data': (self.a.data.mem.ptr, False),
-            'version': 0,
-        }
-        if self.has_strides:
-            desc['strides'] = self.a.strides
-        return desc
+        if self._desc is None:
+            desc = {
+                'shape': self.a.shape,
+                'strides': self.a.strides,
+                'typestr': self.a.dtype.str,
+                'descr': self.a.dtype.descr,
+                'data': (self.a.data.ptr, False),
+                'version': 2,
+            }
+            self._desc = desc
+        return self._desc
+
+    def set_mask(self, mask):
+        if self._desc is None:
+            self.__cuda_array_interface__
```

Comment:
Why is this implemented this way?
How about passing `mask` to `__init__`?

Reply:
See my reply below. Yes, I will make the change.